hax/treemakers/common.py

```diff
@@ -162,6 +162,15 @@ def extract_data(self, event):
                     result['largest_other_s2_pattern_fit'] = getattr(rp, 'goodness_of_fit')
                     break
 
+        # get information for MisIdS1SingleScatter cut
+        result['largest_s2_before_main_s2_area'] = float('nan')
+
+        # grab the largest s2 that occurs in time before the main S2
+        for s2_i in event.s2s:
+            if event.peaks[s2_i].hit_time_mean < s2.hit_time_mean:
+                result['largest_s2_before_main_s2_area'] = event.peaks[s2_i].area
```

Comment:
I don't see how "largest" is calculated in this for loop. I would assume sth like:
all_s2 = events.s2s
candidate_s2 = all_s2[all_s2.hit_time_mean < s2.hit_time_mean]
result['largest_s2_before_main_s2_area'] = candidate_s2.area.max()

Reply:
The s2s are sorted by area right? So the first S2 in the loop that occurs earlier in time than the main S2 will be the largest one. I'll work on a double-check of this.

Reply:
ok, "break" did the trick.